src/Compilers/CSharp/Test/Symbol/Symbols/RequiredMembersTests.cs

```diff
@@ -0,0 +1,1181 @@
+            // (18,20): error CS0106: The modifier 'required' is not valid for this item
+            // required interface I1
+            Diagnostic(ErrorCode.ERR_BadMemberFlag, "I1").WithArguments("required").WithLocation(18, 20),
+            // (20,18): error CS0106: The modifier 'required' is not valid for this item
+            //     required int Prop3 { get; set; }
+            Diagnostic(ErrorCode.ERR_BadMemberFlag, "Prop3").WithArguments("required").WithLocation(20, 18),
+            // (21,18): error CS0525: Interfaces cannot contain instance fields
+            //     required int Field;
+            Diagnostic(ErrorCode.ERR_InterfacesCantContainFields, "Field").WithLocation(21, 18),
+            // (21,18): warning CS0649: Field 'I1.Field' is never assigned to, and will always have its default value 0
+            //     required int Field;
+            Diagnostic(ErrorCode.WRN_UnassignedInternalField, "Field").WithArguments("I1.Field", "0").WithLocation(21, 18),
+            // (29,21): error CS0106: The modifier 'required' is not valid for this item
+            //     required int I2.Prop4 => 0;
+            Diagnostic(ErrorCode.ERR_BadMemberFlag, "Prop4").WithArguments("required").WithLocation(29, 21)
+        );
+    }
+
+    [Fact]
+    public void InvalidModifierCombinations()
+    {
+        var comp = CreateCompilationWithRequiredMembers(@"
+unsafe struct C
+{
+    required const int F1 = 1;
+    required static int F2 = 2;
+    required static int P1 { get; set; }
+    required fixed int F3[10];
+}
+", options: TestOptions.UnsafeReleaseDll);
+
+        comp.VerifyDiagnostics(
+            // (4,24): error CS0106: The modifier 'required' is not valid for this item
+            //     required const int F1 = 1;
+            Diagnostic(ErrorCode.ERR_BadMemberFlag, "F1").WithArguments("required").WithLocation(4, 24),
+            // (5,25): error CS0106: The modifier 'required' is not valid for this item
+            //     required static int F2 = 2;
+            Diagnostic(ErrorCode.ERR_BadMemberFlag, "F2").WithArguments("required").WithLocation(5, 25),
+            // (5,25): warning CS0414: The field 'C.F2' is assigned but its value is never used
+            //     required static int F2 = 2;
+            Diagnostic(ErrorCode.WRN_UnreferencedFieldAssg, "F2").WithArguments("C.F2").WithLocation(5, 25),
+            // (6,25): error CS0106: The modifier 'required' is not valid for this item
+            //     required static int P1 { get; set; }
+            Diagnostic(ErrorCode.ERR_BadMemberFlag, "P1").WithArguments("required").WithLocation(6, 25),
+            // (7,24): error CS0106: The modifier 'required' is not valid for this item
+            //     required fixed int F3[10];
+            Diagnostic(ErrorCode.ERR_BadMemberFlag, "F3").WithArguments("required").WithLocation(7, 24)
+        );
+    }
+
+    [Fact]
+    public void LangVersion()
```

Comment:
The LangVer test should also check using `TestOptions.RegularNext`. All other tests use preview, which will keep sliding forward to C# 12 and beyond.